src/main/java/com/google/devtools/build/lib/actions/cache/CompactPersistentActionCache.java

```diff
@@ -466,6 +467,14 @@ private static void encodeRemoteMetadata(
     VarInt.putVarInt(value.getLocationIndex(), sink);
 
     VarInt.putVarInt(indexer.getOrCreateIndex(value.getActionId()), sink);
+
+    Optional<PathFragment> symlinkTargetExecPath = value.getSymlinkTargetExecPath();
+    if (symlinkTargetExecPath.isPresent()) {
+      VarInt.putVarInt(1, sink);
```

Comment:
nit: Another option is to put a value which cannot be a correct index (e.g. -1) when it is absent. That may ironically be more than 1 byte for -1 (since varint is 1 byte for 0..127). You could reserve 0 and do index + 1, but it is a little weird.

Reply:
I think using 0 and 1 is fine for what's effectively a presence bit.